photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -138,9 +156,9 @@ class CoordinateDescent(
               .setName(s"Updated model with coordinateId $coordinateId at iteration $iteration")
               .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
               .materialize()
-
           case _ =>
         }
+        // We need to split out the following 2 match expressions: [[FactoredRandomEffectModel]] matches both
         oldModel match {
           case broadcastLike: BroadcastLike => broadcastLike.unpersistBroadcast()
           case _ =>
```

Comment:
@ankans I believe it's done this way because we have some objects that are both `BroadcastLike` and `RDDLike` (i.e., they're not mutually exclusive). If they were in the same match statement, only one of them would be matched. That sounds like a strange mixture of traits to me, though, so we should probably evaluate it and make sure it's right.